openapi-cli/src/test/resources/generators/diagnostic_files/single_allOf.yaml

```diff
@@ -0,0 +1,54 @@
+openapi: 3.0.1
+info:
+  title: Sample API
+  description: API description in Markdown.
+  version: 1.0.0
+servers:
+  - url: https://api.example.com
+paths:
+  /users/{id}:
+    get:
+      summary: Returns a list of users.
+      operationId: getUserById
+      description: Optional extended description in Markdown.
+      parameters:
+        - name: id
+          in: path
+          description: ID of user to fetch
+          required: true
+          schema:
+            allOf:
+              - $ref: "#/components/schemas/Name"
+              - description: "Name details"
```

Comment:
Will we need a negative test case to evaluate the output, when an allOf schema is having no valid types? :)

Reply:
Addressed in https://github.com/ballerina-platform/openapi-tools/pull/1521/commits/e77dd15e8ff5a233bf35281b2e335fc87cf01a34 to generate the type as `anydata` in such scenarios